Responsive issues (Fixes #800)

juzen2003 opened the pull request:
- Fixes #800 
- Were any Django, import pipeline, table_schema, or dictionary files modified? N
  - Database used: opus3_test_200211
  - All Django tests pass: Y
- Were any JavaScript or CSS files modified? Y
  - JSHINT run on all affected files: Y
  - Tested on Chrome, Firefox, Safari, and iOS: Y
    (Remember to test all browser sizes)
  - Tested for race conditions (with delayed API calls if applicable): NA

Description of changes:
- In search sidebar and metadata selector menu, when menu items start to wrap, make sure they are aligned with the text, not the (i) icon.
- In search tab, group qtype, (i) icon, and trash icon. They will move to a different line together when browser shrinks.
- In search widgets, group Min:/Max: and their corresponding input tags. Min:/Max: will stay together with their corresponding input tags when browser shrinks and they move to a different line.
- When search widget title starts to wrap, make sure it's aligned with the text instead of the (i) icon.
- In detail tab, group the value and the magnifying glass. They will move together to a different line when browser shrinks.
- Fixed the height of main nav bar when browser width changes.
  - Fixed in https://github.com/SETI/pds-opus/pull/970
- Help menu fit in the screen when browser height is small.
  - Fixed in #852 

Known problems:
- Note: In select metadata menu (different looks in UI):
  - To make the text aligned, we have to set .op-menu-item display to flex. Once this is done, the height of the span tag (.op-menu-item) will be a bit larger, and we have to add margin-top & bottom (1px for now) to create some spacing between each menu item. This will make the height of menu items larger than ones in production sites. 
  - In select metadata menu, when window is narrow, and if a menu item text is wrapped into the 2nd line, the highlighted background will cover the entire row for that menu item. (Match the behavior in search sidebar now)

juzen2003 committed:
Make sure text in the menu align with text in the 1st line instead of
(i) icon when wrapping into the 2nd line.
- In menu.html (line 92,101,103,109), add span tag to wrap the menu text.
  This will be used for css later.
- In opus.css (line 2681-2692), add styling to make sure text in the
  menu align with text in the 1st line instead of (i) icon when wrapping
  into the 2nd line.

juzen2003 committed:
Add spacing between each menu item in select metadata menu in opus.css
(line 768-773).

juzen2003 committed:
Make sure widget title remains lined up with itself and to the right of
(i) icon as the browser shrinks and the title wraps.
- In widget.html (line 12), add bootstrap class to make the div as the
  flex container, this will make sure title text aligned with itself
  when it's wrapped into the 2nd line.
- In widget.html (line 14-24), wrap widget title, unit select, and spinner
  with a div. This will make sure spinner stays right next the widget title
  or unit select.
- In opus.css (line 171-175), add spacing between (i) icon and widget
  title text in search tab.

juzen2003 committed:
Make the (i) icon, qtype dropdown, and trash icon stay together when the
browser gets narrower.
- In widgets.js (line 1332-1341), Wrap (i) icon, qtype, and trash icon
  with a div. This will make sure these three elements stay together when
  browser gets narrow.
- In widgets.js (line 499,501,502), change .parent() to .parents(). This
  will make sure we can properly select the input element from the clicked
  trash icon.

juzen2003 committed:
Make sure input focus out to trigger a change event when clicking any
add/remove input icon.
- In search.js (line 109-114, 120-124), update the condition to call
  preventDefault when clicking insdie a widget body. Ensure input will
  focus out to trigger a change event when clicking any add/remove input
  icon.

juzen2003 committed:
Make min/max and its corresponding input stay together as a unit when they
move to a different line.
- In widgets.js (line 1342-1349), Assign classes to the li of range min/max
  inputs. These classes will be used to add the styling to group min/max
  with its corresponding input tag.
- In opus.css (line 261-281), add the styling to group Min/Max with its
  corresponding input when window is narrow.

juzen2003 committed:
Make sure the magnifying glass always stay with the value in detail page.
- In detail_metadata_internal.html (line 14,16,18,26), wrap value and magnifying
  glass with a span tag.
- In etail_metadata_slugs_internal.html (line 11,12,20), wrap value and
  magnifying glass with a span tag.
- In opus.css (line 218-222), add the styling to group the magnifying
  glass and the value together in detail page.

juzen2003 committed:
Fixed the test case in test_results_api.py, this is because a span tag
with class name equal to op-detail-entry-values is added to wrap the
value and magnifying glass together in detail tab.

rfrenchseti commented on opus/application/apps/results/templates/results/detail_metadata_internal.html:
```diff
@@ -11,17 +11,19 @@
+                                {{ param_info.label_results }}:&nbsp;
                             {% endif %}
+                            <span class="op-detail-entry-values">{{ val }}
                             {% if param_info.is_string_or_mult %}
```
I think this if block should be indented

rfrenchseti commented on opus/application/apps/results/templates/results/detail_metadata_slugs_internal.html:
```diff
@@ -8,14 +8,16 @@
-                    {{ param }}: {{ values.0 }}
+                    {{ param }}:&nbsp;
+                    <span class="op-detail-entry-values">{{ values.0 }}
                     {% if values.1.is_string_or_mult %}
```
I think this if block should be indented

rfrenchseti commented on opus/application/static_media/css/opus.css:
```diff
@@ -249,10 +260,31 @@ Not sure if this is the desired behavior
         width: 100%;
     }
+
+    /* Algin the Min/Max input tag when they are in different lines */
```
Align

rfrenchseti commented on opus/application/static_media/css/opus.css:
```diff
@@ -2671,9 +2710,22 @@ at the right of -- OR -- label. */
 }
 
-/* Make sure label names & hints in wdigets have a caret cursor when users mouseover. */
+/* Make sure label names & hints in wdigets have a caret cursor when users mouseover */
```
widgets

rfrenchseti reviewed (changes requested):
- I don't see the qtype, (i), and trashcan grouped together. The trashcan still moves to the next line by itself. The qtype and (i) are grouped together.
- Also maybe the "+(OR)" should be grouped with the qtype, (i), and trashcan.
- I found one more issue that should be fixed: The rotating chevron (> or v) next to menu category names should stay with the last word so that it never wraps by itself.
- I think this is the correct behavior: "In select metadata menu, when window is narrow, and if a menu item text is wrapped into the 2nd line, the highlighted background will cover the entire row for that menu item. (Match the behavior in search sidebar now)"

juzen2003 committed:
Fix the typo in opus.css and clean up code style (indentation) in
detail_metadata_internal.html & detail_metadata_slugs_internal.html.

juzen2003 committed:
Group qtype, (i) icon, trash icon, and "+" icon so that they will be
wrapped into a different line together.
- In widgets.js (line 390,392), properly append the trash icon inside
  .op-qtype-wrapping-group.
- In widgets.js (line 329-330), properly append the "+" icon inside
  .op-qtype-wrapping-group.

juzen2003 committed:
Update the logic to make sure "+" icon will be wrapped within .op-add-inputs.
- In widgets.js (line 1340-1360), wrapping "+" icon within .op-add-inputs
  after it's been appended in DOM.
- In widgets.js (line 330-334), update the logic in attachAddInputIcon
  to make sure "+" icon is properly attached.

juzen2003 committed:
In menu category, group the last word of category name with triangle
arrow. This will make sure triangle arrow won't move to a different line
alone.
- In menu.js (line 77-89), create wrapTriangleArrowAndLastWordOfMenuCategory
  to wrap the last word of category name and triangle arrow in the menu.
- In menu.js (line 73), selectMetadata.js (line 166), call
  wrapTriangleArrowAndLastWordOfMenuCategory to wrap the last word of
  category name and triangle arrow after the menu is rendered.
- In opus.css (line 350-380), add the styling create triangle arrow and
  make it align with menu category string.
- In menu.html (line 9,31), add the span tag (".op-menu-arrow") inside
  the ".title" span tag. ".op-menu-arrow" is the triangle arrow and will
  be wrapped with the last word of category name later.

juzen2003 committed:
Update the styling for triangle arrow in the menu.
- In opus.css (line 362-371), update the styling for triangle arrow to
  match the one for bootstrap dropdown.

juzen2003 committed:
Update wrapTriangleArrowAndLastWordOfMenuCategory to make sure the spacing
is correct for menu category name.
- In menu.js (line 87-88), update the logic to properly insert a space
  if the name string has more than one word.

juzen2003 committed:
Update the position of triangle arrows in the menu to match ones in the
current production site.

juzen2003 committed:
Merge remote-tracking branch 'upstream/master' into 800_responsive_issues

juzen2003 committed:
Add import json in apps/paraminfo/models.py.
